bindings/python/scripts/parse.py

```diff
@@ -1,145 +1,143 @@
+            parsed_info["access_specifier"] = cursor.access_specifier.name
         if cursor.result_type.spelling != "":
-            holder["result_type"] = cursor.result_type.spelling
+            parsed_info["result_type"] = cursor.result_type.spelling
         if cursor.brief_comment:
-            holder["brief_comment"] = cursor.brief_comment
+            parsed_info["brief_comment"] = cursor.brief_comment
         if cursor.raw_comment:
-            holder["raw_comment"] = cursor.raw_comment
-        holder["members"] = []
-
-        parsed_list.append(holder)
+            parsed_info["raw_comment"] = cursor.raw_comment
+        parsed_info["members"] = []
 
     for child in cursor.get_children():
-        if node_in_this_file(child, this_filename):
-            child_list = []
-            generate_parsed_info(
-                child, filter, lines, this_filename, depth + 1, child_list,
+        if is_node_in_this_file(cursor=child, filename=this_filename):
+            child_parsed_info = generate_parsed_info(
+                cursor=child, this_filename=this_filename, depth=depth + 1,
             )
-            if child_list and parsed_list:
-                if len(parsed_list[0]["members"]):
-                    parsed_list[0]["members"].append(child_list[0])
-                else:
-                    parsed_list[0]["members"] = child_list
-
+            if child_parsed_info and parsed_info:
+                parsed_info["members"].append(child_parsed_info)
 
-def parse_arguments(args):
-    parser = argparse.ArgumentParser(description="C++ libclang parser")
-    parser.add_argument("files", nargs="+", help="The source files to search")
-    return parser.parse_args(args)
+    return parsed_info
 
 
-def get_output_path(source, output_dir):
-    x_list = source.split("pcl/", 1)[-1]
-    x_list = x_list.split("/")
-    extra = ["pcl", "include"]
+def get_compilation_commands(compilation_database_path, filename):
+    """
+    Returns the compilation commands extracted from the the compile command
 
-    filename = x_list[-1].split(".")[0]
-    relative_dir = "/".join(x for x in x_list[:-1] if x not in extra)
-    dir = os.path.join(output_dir, relative_dir)
+    Arguments:
+        - compilation_database_path: The path to `compile_commands.json`
+        - filename: The file's name to get it's compilation commands
 
-    # ensure the new directory exists
-    if not os.path.exists(dir):
-        os.makedirs(dir)
+    Returns:
+        - compilation commands (list): The arguments passed to the compiler
+    """
 
-    return f"{dir}/{filename}.json"
+    compilation_database = clang.CompilationDatabase.fromDirectory(
+        buildDir=compilation_database_path
+    )
+    compilation_commands = compilation_database.getCompileCommands(filename=filename)
+    # extracting argument list from the command's generator object
+    return list(compilation_commands[0].arguments)[1:-1]
```

Comment:
Comments on why are more important. In this case, why choose 0, 1 and -1

Reply:
Updated the function: c6b4ab3.

Same for the comments below.

Reply:
Updated